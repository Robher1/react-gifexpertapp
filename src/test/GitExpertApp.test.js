import React from 'react';
import { shallow } from 'enzyme';
import { GitExpertApp } from '../GitExpertApp';

describe('Pruebas en <GitExpertApp />', ()=>{
    
    test('debe de mostrar el componente correctamente', () => {

        const wrapper = shallow( <GitExpertApp /> );
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorias', async () => {

        const categories = ['mangos', 'sandia'];

        const wrapper = shallow( <GitExpertApp defaultCategories={ categories } /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    });

});
