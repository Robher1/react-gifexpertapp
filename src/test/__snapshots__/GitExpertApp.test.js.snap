// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitExpertApp /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GitExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="mangos"
      key="mangos"
    />
    <GifGrid
      category="sandia"
      key="sandia"
    />
  </ol>
</Fragment>
`;
